### ISSUE TITLE
`get_origin`'s return type is unknown

### ISSUE THREAD

**Issue by Zomatree** [microsoft/pyright#5051]
**Describe the bug**
when using `get_origin` with pyright's strict mode, it errors saying that the return type is unknown, pylance reports the type as `Any` but, `reveal_type` reports the type as `Unknown`

**To Reproduce**
```python
from typing import Any, get_origin

def foo(v: Any):
    x = get_origin(v)

    reveal_type(x)
    print(x)
```
```python
/home/zoma/projects/pyright_issues/repro.py
  /home/zoma/projects/pyright_issues/repro.py:5:5 - error: Type of "x" is unknown (reportUnknownVariableType)
  /home/zoma/projects/pyright_issues/repro.py:7:17 - information: Type of "x" is "Unknown"
  /home/zoma/projects/pyright_issues/repro.py:8:11 - error: Argument type is unknown
```

**Expected behavior**
The type should be `Any` or similar.

**Screenshots or Code**
![Code_-_Insiders_tSKQGwXApw](https://user-images.githubusercontent.com/39768508/235599992-6b6a093c-11cb-48b8-8279-5589ad9f5693.png)

**VS Code extension or command-line**
`pyright 1.1.305`
`Pylance language server 2023.4.41 (pyright c400d889)`

**Comment by erictraut** [comment]
This is intended behavior. The `get_origin` method is overloaded, and the correct overload cannot be determined based on the input arguments (because you are passing `Any`). That means the result cannot be determined, so it is `Unknown` (i.e. an implicit `Any`).

**Comment by Zomatree** [comment]
Surely then, all uses of `Any` should error then should they not? `get_origin` should be not special here. or the return type should be `Any` instead of `Unknown`.

**Comment by erictraut** [comment]
You said that you enabled strict mode. In strict mode, the `reportUnknown...` family of checks are enabled. They report the use of `Unknown`, which is an implicit `Any`. The use of an explicit `Any` is not flagged as an error.

You mentioned that pylance is reporting the type as `Any`. That's probably because you have the typeCheckingMode set to "off" in pylance. When it's "off", all `Unknowns` are displayed as `Any` to avoid confusing people who are not interested in the details of type checking.

**Comment by Zomatree** [comment]
I don't have typeCheckingMode set to "off", otherwise my errors in my screenshots would not be reported.

**Comment by erictraut** [comment]
Oh, you're referring to the inlined type annotation in the pylance screen shot. (I thought you were referring to the hover text, which will display `Unknown` if type checking is enabled but `Any` if it's not.) The inlined type annotations use `Any` because `Unknown` is not a valid type in the Python type system, so if you were to use the code action to add the type annotation to your code, it would produce errors if it were `Unknown`.

The behavior you're seeing here is as designed. The `get_origin` is not special in any way. When you enable strict mode, pyright will report any use of a value with an `Unknown` type. And when you evaluate an overloaded function where the correct overload cannot be determined because one or more arguments are `Any` or `Unknown`, the result of that call will be `Unknown`.
